src/wallet/home_cards/manager.rs

```diff
@@ -76,16 +83,33 @@ impl HomeCardsManager {
     }
 
     /// Handles a deferred deep link by parsing it and saving the generated `HomeCards` to `HomeCardsStorage`.
+    /// `HomeCard::ContinueRadQuest` if found in the link parsing result, replaces `HomeCard::StartRadQuest` if it's stored.
```

Comment:
hmm, this does not seem to be possible - by definition this function should be called only after user did install the wallet, so there cannot be `StartRadQuest` stored. 
Rather, it is the other way around - in `wallet_created`, we should not add the `HomeCard::StartRadQuest` if `HomeCard::ContinueRadQuest` is already present.

Reply:
You're right. What about keeping this logic in both methods to make sure there is no chance both `StartRadQuest` and `ContinueRadQuest` are present at the same time? This way Sargon won't care about the order of the method calls. Also, for the deep link to be resolved the AF SDK performs a network request, then we perform a network request for the dApp icon URL, so there is a tiny possibility the deep link won't be handled fast enough. Keeping this logic also would prevent any scenarios like this.